src/Plugin/CheckImports.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | CheckImport plugin:
+--   This plugin checks if imported identifiers as unqualified
+--   exist and lists them.
+module Plugin.CheckImports
+  ( -- NOTE: The name "plugin" should be used to be called via GHC plugin mechanism.
+    plugin,
+  )
+where
+
+import Control.Monad.IO.Class (MonadIO (liftIO))
+import Data.Char (isAlpha)
+import Data.Foldable (for_)
+import Data.IORef (readIORef)
+import Data.List (foldl', sort)
+import Data.Map (Map)
+import qualified Data.Map as M
+import Data.Set (Set)
+import qualified Data.Set as S
+import GHC.Driver.Session (DynFlags, getDynFlags)
+import GHC.Plugins
+  ( CommandLineOption,
+    ModSummary,
+    Name,
+    Plugin (..),
+    defaultPlugin,
+    localiseName,
+    showSDoc,
+  )
+import GHC.Tc.Types (TcGblEnv (..), TcM)
+import GHC.Types.Name.Reader
+  ( GlobalRdrElt (..),
+    GreName (..),
+    ImpDeclSpec (..),
+    ImportSpec (..),
+  )
+import GHC.Unit.Module (ModuleName)
+import GHC.Utils.Outputable (Outputable (ppr))
+import Prelude hiding ((<>))
+
+plugin :: Plugin
+plugin =
+  defaultPlugin
+    { typeCheckResultAction = typecheckPlugin
+    }
+
+printPpr :: (Outputable a, MonadIO m) => DynFlags -> a -> m ()
+printPpr dflags = liftIO . putStrLn . showSDoc dflags . ppr
+
+formatName :: DynFlags -> Name -> String
+formatName dflags name =
+  let str = showSDoc dflags . ppr . localiseName $ name
+   in case str of
+        (x : _) ->
+          if isAlpha x
+            then str
+            else "(" ++ str ++ ")"
```

Comment:
Kinda strange that `showSDoc` doesn't do this for you already.

Reply:
Ah, this is because I wanted to let users directly copy&paste the resultant report to the code. So operators should be wrapped by parenthesis. The general purpose of `showSDoc` is to show the text, without knowing any context about operators, so this ad hoc treatment is done on my side. I will add a comment about this in a following commit. Thanks for pointing this out.